src/docfx/SubCommands/MetadataMerger.cs

```diff
@@ -103,6 +108,43 @@ private static void MergeToc(MetadataMergeParameters parameters, string outputBa
                 YamlMime.TableOfContent);
         }
 
+        private void InitMetaTable(FileModel model)
+        {
+            var content = model.Content as PageViewModel;
+            if (content != null && model.Uids.Length > 0)
+            {
+                _metaTable.Add(model.Uids.First().Name, content.Metadata);
```

Comment:
big difference in space complexity. nice catch!

---
In reply to: [90815670](https://github.com/dotnet/docfx/pull/941#discussion_r90815670) [](ancestors = 90815670,90814873)